### PR TITLE
Board resource

### PR DESCRIPTION
**Pull request by fitzoh**
TODO: Documentation

### CODE REVIEW

File: honeycombio/resource_board_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        query_json = data.honeycombio_query.test[0].json
+    }
+      
+    query {
+        caption = "test query 1"
+        dataset = "%s"
+        query_json = data.honeycombio_query.test[1].json
+    }
+
+}`, dataset, dataset)
```
Comment (fitzoh) [inline honeycombio/resource_board_test.go:62]: This part look reasonable to you?

Comment (kvrhdn) [reply honeycombio/resource_board_test.go:62]: Yeah, I think this is great. This seems intuitive and easy to use.
I had put something similar here: https://github.com/kvrhdn/terraform-provider-honeycombio/issues/3#issue-669230480

File: honeycombio/resource_board.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                            Type:     schema.TypeString,
+                            Required: true,
+                        },
+                        "query_json": {
+                            Type:     schema.TypeString,
+                            Required: true,
+                        },
+                    },
+                },
+                Optional: true,
```
[the remark this answers is not in the document]
Comment (fitzoh) [reply honeycombio/resource_board.go:49]: 1f1ae8312a35a61c6c62b720d354fbd043ebdb2b

File: honeycombio/resource_board.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    d.SetId(t.ID)
+    return resourceTriggerRead(d, meta)
+}
+
+func resourceBoardDelete(d *schema.ResourceData, meta interface{}) error {
+    client := meta.(*honeycombio.Client)
+
+    dataset := d.Get("dataset").(string)
+
+    return client.Triggers.Delete(dataset, d.Id())
```
[the remark this answers is not in the document]
Comment (fitzoh) [reply honeycombio/resource_board.go]: thing I got the rest of those in 1012b9b4df528322a259de5dee2f96d202e57167

File: honeycombio/resource_board.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        return err
+    }
+
+    t, err = client.Triggers.Update(dataset, t)
+    if err != nil {
+        return err
+    }
+
+    d.SetId(t.ID)
+    return resourceTriggerRead(d, meta)
```
[the remark this answers is not in the document]
Comment (fitzoh) [reply honeycombio/resource_board.go]: ThinK i got the rest of those in 1012b9b4df528322a259de5dee2f96d202e57167

File: honeycombio/resource_board.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            },
+            "description": {
+                Type:     schema.TypeString,
+                Optional: true,
+            },
+            "style": {
+                Type:     schema.TypeString,
+                Optional: true,
+                Default:  "list",
+            },
```
[the remark this answers is not in the document]
Comment (fitzoh) [reply honeycombio/resource_board.go]: 520695569bf4a9a9299a13be95f1e12a52fda8b5

File: honeycombio/resource_board.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+                            Optional: true,
+                        },
+                        "dataset": {
+                            Type:     schema.TypeString,
+                            Required: true,
+                        },
+                        "query_json": {
+                            Type:     schema.TypeString,
+                            Required: true,
+                        },
```
[the remark this answers is not in the document]
Comment (fitzoh) [reply honeycombio/resource_board.go]: 6428744cdf507243d57f556daba10adf41ceebca

**Comment by fitzoh** [comment]
I added a board example in 431d865a2e05ca597d0c50052626e4123507681b and discovered a fun issue with the query_spec data source.

If the filter op is `exists` or `does-not-exist`, the honeycomb API returns an error if you send any `value`, even an empty string.

Updated some code and added a test case in a387bad21ad50cb19baa67bffbda091222c325c8 to check for that.
I'm hoping the V2 SDK (https://github.com/kvrhdn/terraform-provider-honeycombio/issues/9) allows writing a validation function for that, but I wasn't seeing a way to do so now.

File: honeycombio/resource_board.go [hunk 7]
```diff
@@ -0,0 +1,154 @@
+package honeycombio
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	honeycombio "github.com/kvrhdn/go-honeycombio"
```
Comment (kvrhdn) [inline honeycombio/resource_board.go:9]: A small formatting issue, goimports expects imports to be grouped by stdlib / other packages.
```suggestion
	"encoding/json"
	"errors"

	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
	honeycombio "github.com/kvrhdn/go-honeycombio"
```

File: example/board/main.tf [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+resource "honeycombio_board" "board" {
+  name        = "Request percentiles"
+  description = "${join(", ", local.percentiles)} of all requests for ThatSpecialTenant for the last 15 minutes."
+  style       = "list"
+
+
+  //Use dynamic config blocks to generate a query for each of the percentiles we're interested in
+  dynamic "query" {
```
[the remark this answers is not in the document]
Comment (fitzoh) [reply example/board/main.tf:43]: yeah, it was new in 0.12: https://www.hashicorp.com/blog/hashicorp-terraform-0-12-preview-for-and-for-each/

**Comment by fitzoh** [comment]
>I think, except for the imports formatting, this is ready to merge 👍🏻

👍 

>I particularly like how easy it is to share queries between triggers and boards thanks to honeycombio_query and query_json 🙂

Agreed

>We'll have to add a documentation page before releasing this, but I don't mind writing that since I did the others as well already.

Sounds good to me

**Pull request merged**